.github/workflows/main.yml

```diff
@@ -0,0 +1,153 @@
+name: A workflow for updating discovery artifacts
+# Controls when the action will run.
+
+on:
+  # Triggers the workflow on pull request events but only for the master branch
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Update Discovery Artifacts PR
+    runs-on: ubuntu-latest
+    if: ${{github.event.pull_request.user.login == 'parthea' && github.event.pull_request.title == 'Add github action to update discovery artifacts'}}
+    steps:
+      - name: Check out working directory for changes
+        uses: actions/checkout@v2
+        with:
+          ref: refs/heads/master
+          path: ./branch
+          fetch-depth: '0'
```

Comment:
I think its default clone might be shallow? might be worth checking.

Reply:
Good catch! I've removed `fetch-depth`